openlibrary/catalog/utils/__init__.py

```diff
@@ -179,7 +174,7 @@ def match_with_bad_chars(a, b):
         return True
 
     def drop(s):
-        return re_drop.sub('', six.ensure_text(s))
+        return re_drop.sub('', s.decode() if isinstance(s, bytes) else s)
```

Comment:
This one looks a touch confusing; can you expand on what's happening?

Reply:
https://github.com/benjaminp/six/blob/master/six.py#L939

Reply:
Lgtm! It looks like it's using the default decode params there as well. Only difference is this won't error if `s` is non-str AND non-bytes, but that's probably ok?